### PR TITLE
Update NA repr

### LINKED ISSUE #30415: DISCUSS: disambiguation of NA and "NA" in reprs
## Change to `str` `dtype` behaviour for missing elements

Following comments the discussion about how to handle missing `NA` scalar values in https://github.com/pandas-dev/pandas/issues/28778 I was asked to raise my question as this seperate issue. 

My rather prosaic question is how if missing `str` elements are given the value `NA`, how would I distinguish between a missing `str` value and the two-character string 'NA'?

I ask as NA is a common abbreviation for 'Not Applicable', 'North America' et al, in a way that in my experience that 'NaN' or 'Not a Number' isn't

That is, if 'NA' were generated as the default missing `str` `dtype` value, especially if introduced as change rather than as a opt-in, it risks becoming a UX developer issue as I (for one) would no longer know if 'NA' is a valid or a missing data value.

For what it's worth, current idiomatic behaviour is that in a missing values would be replaced by `None`   `dtype`:
```python
   >>> array = [['No-one', 'Nadie'], ['Expects']]
   >>> df = pd.DataFrame(array, columns=['En', 'Es'])
           En     Es
   0   No-one  Nadie
   1  Expects   *None*
```

The `dtypes` here are:
```python
   >>> [type(i) for i in df['Es']]
   [<class 'str'>, <class 'NoneType'>]
```

Given this, my thought is that `NA` is not a suitable  *default* replacement for missing `str` `dtype` elements rather `None` of `NoneType` `dtype`

### PR DESCRIPTION
**Pull request by TomAugspurger**
Closes https://github.com/pandas-dev/pandas/issues/30415

```python

In [2]: df = pd.DataFrame({"A": pd.array([1, 2, None])})

In [3]: df
Out[3]:
      A
0     1
1     2
2  <NA>

In [4]: df.A
Out[4]:
0       1
1       2
2    <NA>
Name: A, dtype: Int64

In [5]: df.A.array
Out[5]:
<IntegerArray>
[1, 2, <NA>]
Length: 3, dtype: Int64

```

I think adding color with ANSI codes / custom HTML formatting is still worth doing as in #30778, but this is an improvement for now.

### CODE REVIEW

File: pandas/_libs/missing.pyx [hunk 1]
```diff
@@ -354,10 +354,7 @@ class NAType(C_NAType):
         return NAType._instance
 
     def __repr__(self) -> str:
-        return "NA"
-
-    def __str__(self) -> str:
```
Comment (TomAugspurger) [inline pandas/_libs/missing.pyx]: Python will automatically use `__repr__` if one isn't defined, so this was redundant.

**Comment by TomAugspurger** [comment]
I had to include `<NA>` in the list of default NA values for read_csv to get this working. Probably OK to do. Also maybe simplified ExtensionBlock.to_native_types, but may have broken something along the way.

**Comment by WillAyd** [comment]
On board with this once green

**Comment by jreback** [comment]
some doc-test failures, otherwise lgtm.

File: doc/source/whatsnew/v1.0.0.rst [hunk 2]
```diff
@@ -575,6 +575,7 @@ Other API changes
   Supplying anything else than ``how`` to ``**kwargs`` raised a ``TypeError`` previously (:issue:`29388`)
 - When testing pandas, the new minimum required version of pytest is 5.0.1 (:issue:`29664`)
 - :meth:`Series.str.__iter__` was deprecated and will be removed in future releases (:issue:`28277`).
+- Added ``<NA>`` to the list of default NA values for :meth:`read_csv` (:issue:`30821`)
```
Comment (jreback) [inline doc/source/whatsnew/v1.0.0.rst]: may want to also mention in the NA section

**Comment by jreback** [comment]
I think you need to update the tests if a nullable integer / bool / string array appears in Series/DataFrame? (or does this not work yet)?

**Comment by TomAugspurger** [comment]
It works. We have smoke tests in the base class. Will add some dedicated ones.

**Comment by TomAugspurger** [comment]
Passing now. Merging in an hour.

**Comment by jorisvandenbossche** [comment]
> I had to include <NA> in the list of default NA values for read_csv to get this working. 

Can you explain why? Pandas does not write such format to CSV files, so it shouldn't be needed for a roundtrip

File: pandas/io/formats/format.py [hunk 3]
```diff
@@ -1230,7 +1230,7 @@ def _format(x):
                     if x is None:
                         return "None"
                     elif x is NA:
-                        return "NA"
+                        return formatter(x)
```
Comment (jorisvandenbossche) [inline pandas/io/formats/format.py]: This means that the `formatter` function needs to be able to handle NAs, which now is maybe not the case?

Comment (jorisvandenbossche) [reply pandas/io/formats/format.py]: Will do a PR for this, I think this breaks geopandas

**Pull request merged**